src/ray/util/container_util.h

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <map>
+#include <set>
+#include <sstream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "ray/util/logging.h"
+
+namespace ray {
+
+template <typename T>
+std::string debug_string(const T &obj) {
+  std::stringstream ss;
+  ss << obj;
+  return ss.str();
+}
+
+template <typename... Ts>
+std::string debug_string(const std::pair<Ts...> &pair) {
+  std::stringstream ss;
+  ss << "(" << debug_string(pair.first) << ", " << debug_string(pair.second) << ")";
+  return ss.str();
+}
+
+template <typename C>
+std::string _container_debug_string(const C &c) {
+  std::stringstream ss;
+  ss << "[";
+  for (auto it = c.begin(); it != c.end(); ++it) {
+    if (it != c.begin()) {
+      ss << ", ";
+    }
+    ss << debug_string(*it);
+  }
+  ss << "]";
+  return ss.str();
+}
+
+template <typename... Ts>
+std::string debug_string(const std::vector<Ts...> &c) {
+  return _container_debug_string(c);
+}
+template <typename... Ts>
+std::string debug_string(const std::set<Ts...> &c) {
+  return _container_debug_string(c);
+}
+template <typename... Ts>
+std::string debug_string(const std::unordered_set<Ts...> &c) {
+  return _container_debug_string(c);
+}
+template <typename... Ts>
+std::string debug_string(const std::map<Ts...> &c) {
+  return _container_debug_string(c);
+}
+template <typename... Ts>
+std::string debug_string(const std::unordered_map<Ts...> &c) {
+  return _container_debug_string(c);
+}
+
+template <typename C>
+const typename C::mapped_type &map_find_or_die(const C &c,
```

Comment:
@iycheng i remember you are trying to introduce something similar, but I can't find it in the codebase.

Reply:
Yea, I didn't find anything, but if it's already there, I'm happy to remove mine.